Sources/App/Commands/TriggerBuilds.swift

```diff
@@ -203,18 +203,20 @@ func triggerBuilds(on database: Database,
         .flatten(on: database.eventLoop)
     }
 
-    guard Current.random(0...1) < Current.buildTriggerDownscaling() else {
-        logger.info("Build trigger downscaling in effect - skipping builds")
-        return database.eventLoop.future()
-    }
-
     return Current.getStatusCount(client, .pending)
         .and(Current.getStatusCount(client, .running))
         .flatMap { (pendingJobs, runningJobs) in
             AppMetrics.buildPendingJobsCount?.set(pendingJobs)
             AppMetrics.buildRunningJobsCount?.set(runningJobs)
             var newJobs = 0
             return packages.map { pkgId in
+                let allowListed = Current.buildTriggerAllowList().contains(pkgId)
```

Comment:
I initially intended to make the allow list URL based but it turns out we only have package IDs available at this point. So we'll have to seed the allow list with IDs instead.

Since this list is quite nitty-gritty and should see few changes once updated (and package IDs are stable once packages are added), that should be ok for a feature that's to be used in the `dev` environment only.

One advantage this has is that there's not guessing if it's `http://github.com/foo/bar`, `.../bar.git`, `Bar.git`, `https://...` etc but admittedly the initial lookup is a bit more awkward.

As a reminder: the package id is listed in the HTML source of a package page in a comment, so it's easy to look up even without database access.

<img width="1282" alt="CleanShot 2022-11-25 at 16 32 17@2x" src="https://user-images.githubusercontent.com/65520/204016785-e18dcc33-ba59-4fbc-9a98-68e1b7ddd9ba.png">

Reply:
(We could of course add some sort of lookup from `URL` → `Package.Id` inside build trigger but that's quite a bit of hassle and would pretty much force a conversion over to async/await as a first step, which I want to avoid at this point.)